service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapper.java

```diff
@@ -96,12 +107,34 @@ private String mapDefaultNotImplemented(Resource resource) {
             resource.getIdElement().getIdPart());
     }
 
+    private String mapAllergyIntolerance(Resource resource) {
+        return allergyStructureMapper.mapAllergyIntoleranceToAllergyStructure((AllergyIntolerance) resource);
+    }
+
     private String mapCondition(Resource resource) {
         return conditionLinkSetMapper.mapConditionToLinkSet((Condition) resource, IS_NESTED);
     }
 
-    private String mapProcedureRequest(Resource resource) {
-        return diaryPlanStatementMapper.mapDiaryProcedureRequestToPlanStatement((ProcedureRequest) resource, IS_NESTED);
+    private String mapDocumentReference(Resource resource) {
+        return documentReferenceToNarrativeStatementMapper.mapDocumentReferenceToNarrativeStatement((DocumentReference) resource);
+    }
+
+    private String mapImmunization(Resource resource) {
+        return immunizationObservationStatementMapper.mapImmunizationToObservationStatement((Immunization) resource, IS_NESTED);
+    }
+
+    private String mapListResource(Resource resource) {
+        ListResource listResource = (ListResource) resource;
+
+        if (listResource.hasEntry() && hasCode(listResource.getCode(), COMPONENTS_LISTS)) {
```

Comment:
Are there lists of other codes that we're not yet mapping, or would another list indicate garbage in?

Reply:
There are the only lists we're concerned with. It is the consultation/topic/heading list for structured consultations which are flattened for EHR output.